src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -261,6 +272,53 @@ public Guid(string g)
             this = result.ToGuid();
         }
 
+        /// <summary>Gets the value of the variant field for the <see cref="Guid" />.</summary>
+        /// <remarks>
+        ///     <para>This corresponds to the most significant 4 bits of the 8th byte: 00000000-0000-0000-F000-000000000000. The "don't-care" bits are not masked out.</para>
+        ///     <para>See RFC 9562 for more information on how to interpret this value.</para>
+        /// </remarks>
+        public int Variant => _d >> 4;
+
+        /// <summary>Gets the value of the version field for the <see cref="Guid" />.</summary>
+        /// <remarks>
+        ///     <para>This corresponds to the most significant 4 bits of the 6th byte: 00000000-0000-F000-0000-000000000000.</para>
+        ///     <para>See RFC 9562 for more information on how to interpret this value.</para>
+        /// </remarks>
+        public int Version => _c >> 12;
```

Comment:
Does the sign extension here matter?

Reply:
Yes, since it's the upper 4 most bits. It makes a difference if we return `15` or `-1`, for example.

```suggestion
        public int Version => _c >>> 12;
```